app/controllers/exercises_controller.rb

```diff
@@ -315,9 +315,11 @@ def implement # rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedC
       # we are just acting on behalf of a single user who has already worked on this exercise as part of a programming group **in the context of the current study group**
       session[:pg_id] = pg.id
       @current_contributor = pg
-    elsif session[:pg_id].blank? && PairProgramming23Study.participate_in_pp?(current_user, @exercise) && PairProgramming23Study::FORCED_EXERCISE_IDS.include?(@exercise.id)
+    # TODO: When the user is changing the url the session parameter is not set for the exercise
```

Comment:
I think it is only possible to get the value `pair_programming` for an exercise if we store it in the db or have it in a separate file. I think it is not wanted that we store it in the exercise db because the value can be different for different courses.

Maybe we shouldn't allow normal users to change the page via the URL and therefore just stay on the page when `session[:pg_id].blank?` is true. Then of course we can't save events for the different cases. What do you think?

Reply:
> I think it is not wanted that we store it in the exercise db because the value can be different for different courses.

Indeed, that's why we pass the option through LTI.

> Maybe we shouldn't allow normal users to change the page via the URL and therefore just stay on the page when `session[:pg_id].blank?` is true. 

Sure, this is something we could consider (and might have positive consequences). However, for now, I would not suggest extending this beyond the original scope of this PR. Therefore, I would go with a similar solution to what we had so far.

<hr>

That having said, I think the solution currently in the code is okay. Sure, there is still the issue that the session parameter could still be circumvented (e.g., by starting an exercise without mandatory Pair Programming and then switching the URL to a mandatory Pair Programming exercise). However, a user cannot receive any points through this manipulation (but just see the exercise). Hence, I would accept this for now.

In that regard, I am not sure whether I get the code comments correctly. Did I answer that with my comment already or is there another question I missed (and didn't get)?

Reply:
You answered all questions. I overlooked that the user then cannot get points. Then it should be ok like this for now.